### ISSUE TITLE
[clang-tidy][false-negative] `modernize-raw-string-literal` does not check function-like macro parameters

### ISSUE THREAD

**Issue by tearfur** [llvm/llvm-project#133618]
String literals that are passed in as parameters of a function-like macros are not caught by `modernize-raw-string-literal`. Checked the documentation for `modernize-raw-string-literal` and AFAICT there is no `IgnoreMacros` option or anything in that nature.

https://godbolt.org/z/edvqhdff4

```
<source>:11:18: warning: escaped string literal can be written as a raw string literal [modernize-raw-string-literal]
   11 |     std::cout << "\\\\127.0.0.1\\ADMIN$\\System32"sv << '\n';
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                  R"(\\127.0.0.1\ADMIN$\System32)"sv
1 warning generated.
```

Current trunk: 9747bb182f430bb1bd3525b7f42e88df626e28e5

**Comment by vbvictor** [comment]
Yes, there is actually a FIXME in test file that covers this case.
https://github.com/llvm/llvm-project/blob/057b3c245234e9d9fa3a25fbf04d20e7f3fe2cff/clang-tools-extra/test/clang-tidy/checkers/modernize/raw-string-literal.cpp#L106-L108

**Comment by carlosgalvezp** [comment]
That unit test will not work though, because the string to replace is too short, i.e. the replacement is longer than the original code.
